subsys/jwt/Kconfig

```diff
@@ -7,27 +8,74 @@ menuconfig JWT
 	help
 	  Enable creation of JWT tokens
 
+if JWT
+
 choice
 	prompt "JWT signature algorithm"
 	default JWT_SIGN_RSA
-	depends on JWT
 	help
 	  Select which algorithm to use for signing JWT tokens.
 
 config JWT_SIGN_RSA
 	bool "Use RSA signature (RS-256)"
-	depends on CSPRNG_ENABLED
-	select MBEDTLS
-	select MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
-	select PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY if PSA_CRYPTO_CLIENT
-	select PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_IMPORT if PSA_CRYPTO_CLIENT
 
 config JWT_SIGN_ECDSA
 	bool "Use ECDSA signature (ES-256)"
+
+endchoice
+
+choice
+	default JWT_USE_PSA
+	prompt "Select crypto library to be used"
+
+config JWT_USE_PSA
+	bool "PSA crypto API library"
+
+config JWT_USE_LEGACY
+	bool "Legacy library: TinyCrypt for ECDSA, Mbed TLS for RSA"
+
+endchoice
+
+# Hidden Kconfigs
+
+config JWT_SIGN_ECDSA_PSA
+	bool
+	default y
+	depends on JWT_SIGN_ECDSA && JWT_USE_PSA
+	select MBEDTLS if !BUILD_WITH_TFM
+	select MBEDTLS_PSA_CRYPTO_C if !BUILD_WITH_TFM
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT
+	select PSA_WANT_ALG_ECDSA
+	select PSA_WANT_ECC_SECP_R1_256
+	select PSA_WANT_ALG_SHA_256
+
+config JWT_SIGN_ECDSA_LEGACY
```

Comment:
Why not a choice here as well? Would make sense to me. Especially since you are using `default y` for both configs, with incompatible conditions. Just introduce a `choice`, remove those `default y`, keeping the `depends on` conditions. No need to even put a `default` in the `choice`.